internal/utils/cdx.go

```diff
@@ -0,0 +1,29 @@
+package utils
```

Comment:
can we make this `cdx_utils`? Ideally the consumer code should look something like:
```
comps := cdx_utils.DiscoverComponents(bom)
```